telemetry/specs/INT.mdk

```diff
@@ -1409,57 +1397,158 @@ INT-MX Header:
-
-  []{tex-cmd: "\newpage"}
-
-  If the *Domain Specific ID* does not match any Domain ID known to this node, then
-  the transit node is required to either:
+  per-hop metadata length (*Hop ML*) set by the INT source, as described below.
 
-  - Send the metadata corresponding to *Instruction Bitmap* and ensure that
-    *DSMdbits* is not set in the Telemetry Report and *MD Length* field reflects the length
-    of the metadata sent in the Telemetry Report [^telem-report], or
-
-  - Send the metadata corresponding to *Instruction Bitmap* and pad the node's INT Metadata
-    with the special all-ones reserved value for a Domain Specific Metadata length,
-    calculated by subtracting from the *Hop ML* a length computed from all bits in the
-    16-bit *Instruction Bitmap*, or
-
-  - Skip INT processing altogether and not send any metadata the monitoring systems.
+* **Optional Domain Specific 'Source-Only' Metadata:** The metadata
+  corresponding to 'source-only' *DS Instruction* bits follows the
+  *DS Instruction* and the *DS Flags* fields in the INT-MX header. The length
+  of this field must be counted in the INT length field of the INT shim header.
 
 * Each INT Transit node along the path that supports INT-MX sends its own metadata
-  values as specified in the *Instruction Bitmap* and *DS Instruction*.
-
-  - If a node is unable to provide a metadata value specified in the
-    instruction bitmap because its value is not available, it must either:
-
-      - ensure that the corresponding bit in *RepMdbits* and/or *DSMdbits* is not set
-        in the Telemetry Report [^telem-report], or
+  values as specified in the *Instruction Bitmap* and *DS Instruction*. Either
+  one of the following two procedures may be implemented:
+
+  1. Modify *RepMdBits* and *DSMdbits* in the Telemetry Report [^telem-report],
+     after copying from INT-MX *Instruction Bitmap* and *DS Instruction*
+     respectively, and recompute *MD Length*.
+
+     - If INT-MX *Instruction Bitmap* bit 0 was set, clear *RepMdBits* bit 0.
+
+     - If a node is unable to provide a metadata value specified in the
+       instruction bitmap because its value is not available, or because it
+       corresponds to a reserved bit, the node must ensure that the
+       corresponding bit in *RepMdbits* and/or *DSMdbits* is not set in the
+       Telemetry Report sent to the monitoring system.
+
+     - If the *Domain Specific ID* does not match any Domain ID known to this
+       node, then the transit node is required to either:
+       - Send the metadata corresponding to *Instruction Bitmap* and ensure that
+         *DSMdbits* is not set in the Telemetry Report and the *MD Length* field
+         reflects the length of the metadata sent in the Telemetry Report, or
+       - Skip INT processing altogether and not send any metadata to the
+         monitoring systems.
+
+     - The INT node should make every effort to include 'source-only' metadata
```

Comment:
The first level bullet is only transit, but this really applies to source, transit, and sink. Up in the DS Instruction definition, it does say:

> each node including the source, transit, and sink nodes should report that source metadata to the monitoring system along with the node's own metadata

The details are only defined here for transit. I don't really want to remove "(that was received in the INT-MX header)" since it helps reiterate the context.

I don't really want to move this text somewhere else since I like all the transit node's bit manipulations to be described together.